investor_wallet_platform_base/models/partner.py

```diff
@@ -357,19 +360,38 @@ def get_membership(self, structure):
     def generate_mail_templates(self):
         self.ensure_one()
         if self.mail_serveur_out:
-            mail_templ = self.env['mail.template']._get_email_template_dict()
+            mail_templates = self.env['mail.template']._get_email_template_dict()
 
-            if not self.mail_template_ids:
-                for mt_key, mt_xml_id in mail_templ.items():
+            for mt_key, mt_xml_id in mail_templates.items():
+                template_exists = (
+                    self.env['mail.template']
+                        .get_email_template_by_key(mt_key, self, raise_=False))
+
```

Comment:
You are making a database call for each key of the dict...

better go for
structure_template_keys = self.mail_template_ids.mapped('template_key)

diff = set(mail_templates.keys()) - set(structure_template_keys)

Reply:
Good suggestion, see the changes.